overlord/state/state.go

```diff
@@ -227,6 +236,12 @@ func (s *State) UnmarshalJSON(data []byte) error {
 	s.lastTaskId = unmarshalled.LastTaskId
 	s.lastLaneId = unmarshalled.LastLaneId
 	s.lastNoticeId = unmarshalled.LastNoticeId
+	// Update the last notice timestamp if the one saved to disk is later.
+	// The timestamp on disk does not necessarily reflect the most recent
+	// timestamp of notices which are not stored in state, so the other notice
+	// backends should similarly update the last notice timestamp during
+	// startup.
+	s.HandleReportedLastNoticeTimestamp(unmarshalled.LastNoticeTimestamp)
```

Comment:
I think the comment needs to explain more, we can do this because when we compute the new timestamp for a state notice we are also holding the state lock?